### ISSUE TITLE
Plan Collaborators With "Read only" Permissions Are Still Able to Add Contributors to a Plan

### ISSUE THREAD

**Issue by aaronskiba** [DMPRoadmap/roadmap#3439]
Please complete the following fields as applicable:

**What version of the DMPRoadmap code are you running? (e.g. v2.2.0)**
`v4.2.0`

**Expected behaviour:**
- I'm not 100% sure. However, I'd assume that collaborators with "read only" access on a plan shouldn't be able to modify it in any way.

**Actual behaviour:**
- Plan collaborators with "read only" can successfully add a contributor to a plan.
**Steps to reproduce:**
- 1. Add a collaborator and give them "read only" access
- <img width="1150" alt="Screenshot 2024-07-11 at 10 31 01 AM" src="https://github.com/portagenetwork/roadmap/assets/71047780/a2b67bee-55e7-4aa5-8bdb-0a3674f446b4">
- 2. Sign into the app as the added "read only" user and navigate to the corresponding plan
- <img width="1247" alt="Screenshot 2024-07-11 at 10 31 54 AM" src="https://github.com/portagenetwork/roadmap/assets/71047780/572593a0-3cf3-43d5-8bed-f034383e2b5d">
- 3. Try adding a contributor to the plan
- <img width="1202" alt="Screenshot 2024-07-11 at 10 32 17 AM" src="https://github.com/portagenetwork/roadmap/assets/71047780/d92e385f-baf3-4312-9e84-17d9f48633c8">

**Comment by aaronskiba** [comment]
@martaribeiro is this in fact unwanted behaviour?

**Comment by martaribeiro** [comment]
I think it was @briri who added this feature to Roadmap. Maybe he had a reason to allow a "read only" user to add a contributor to a plan.
I asked Diana and Magdalena to see if there was a reason for this. We are not sure either. To me looks like a bug as John said.

**Comment by briri** [comment]
Yes, sounds like a bug. I can't think of a scenario where a read-only account would need to add contributors
